### ISSUE TITLE
When Serializing a Type Safe Enum to Xml the values are not output

### ISSUE THREAD

**Issue by Porkstone** [mikeobrien/Bender#35]
The structure of the Xml is fine but the node values are missing.

There is a gist here:
https://gist.github.com/anonymous/e6faa9e80b5fa534528d

**Comment by mikeobrien** [comment]
It's going to treat `ApplicantType` as an object (Since it's not a simple type). And since it has no serializable properties or fields there would be nothing to serialize. You can accomplish what you want to do by creating a writer a la:

```csharp
public static string ApplicantXml(Applicant applicant)
{
    return Serialize.Xml(applicant, x => x.Serialization(y => y.PrettyPrint()
        .AddWriter<ApplicantType>((value, source, target, options) => value.ToString())));
}
```

So do other serializers recognize how to serialize this? Do they just automatically call `ToString()` if there are no serializable properties or fields?

**Comment by Porkstone** [comment]
Thanks for the tip.
I'm not sure what the other xml seralizers do but I will try the .net one and add a comment when I find out

**Comment by mikeobrien** [comment]
Yeah, would be curious if they do. That might be a handy option to have. When I get a sec I'll see how hard it would be to implement.

**Comment by mikeobrien** [comment]
So took a quick stab at this and it looks like it will be tricky to implement. One thing I could right now is add is a convenience method to treat certain types as value types during serialization (Basically would do what I noted above). Something along the lines of:

```csharp
public static string ApplicantXml(Applicant applicant)
{
    return Serialize.Xml(applicant, x => x.Serialization(y => 
        y.PrettyPrint().AsSimpleType<ApplicantType>()));
}
```

Would that be helpful?

**Comment by mikeobrien** [comment]
I added that convience method in ddf2a9fd938d41382aa8b86aa49a6fbb86f3b2d9 since it was so easy to add.

**Comment by mikeobrien** [comment]
Published to nuget: [3.0.9](https://www.nuget.org/packages/Bender/)

**Comment by Porkstone** [comment]
Awesome thanks

**Comment by Porkstone** [comment]
I had a play around with the dotnet serializer. At first it complained that the type safe enum class did not have a parameter less construct so I had to add a public parameter less ctor to make it work.
Once I had got is working the dotnet serializer default behavior was to output an empty element. I'm going to spend a bit more time looking at the dotnet serializer to see if I can figure out how to get it to populate the element.
I put my code up here https://github.com/Porkstone/TestXmlSerialisation
